### PR TITLE
test coverage improvement 03

### PR DESCRIPTION
**Pull request by isseis**
This pull request adds comprehensive new tests to improve coverage and robustness for logging, error handling, Slack notifications, and privilege metrics. The changes focus on verifying edge cases, error types, Slack handler behaviors, and privilege metrics calculations, ensuring that these components handle a wide variety of scenarios correctly and do not panic under unexpected conditions.

**Logging and Message Formatting:**
* Added extensive edge case tests for `FormatRecordInteractive` in `message_formatter_test.go`, covering empty, special, mixed, nested, and skipped attributes, as well as empty values, to ensure correct formatting and resilience.

**Error Handling:**
* Added tests for all `PreExecutionError` types, error wrapping/unwrapping, and string formatting in `pre_execution_error_test.go`, verifying that error handling logic is robust and does not panic.

**Slack Notification Handler:**
* Added tests for Slack webhook URL retrieval, handler enablement by log level, conditional notification sending, message formatting for various types, retry logic on failures, and backoff interval generation in `slack_handler_test.go`. These tests use mock servers to validate integration and error handling. [[1]](diffhunk://#diff-1441a1d257bdb608e4ddaa9ba9d9c7f5a2450639dedf78d286e1e33019b77841R4-R12) [[2]](diffhunk://#diff-1441a1d257bdb608e4ddaa9ba9d9c7f5a2450639dedf78d286e1e33019b77841R361-R725)

**Privilege Metrics:**
* Added thorough tests for privilege elevation metrics, covering all scenarios for success rate calculation, including initial state, all successes, all failures, mixed results, and reset behavior in `metrics_test.go`.

### CODE REVIEW

File: test/security/hash_bypass_test.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+	symlinkPath := filepath.Join(tempDir, "symlink.txt")
+	err = os.Symlink(sensitiveFile, symlinkPath)
+	if err != nil {
+		t.Skipf("Cannot create symlinks: %v", err)
+	}
+	defer os.Remove(symlinkPath)
+
+	// Verify symlink - behavior depends on implementation
+	err = validator.Verify(symlinkPath)
+	t.Logf("Symlink verification handled: %v", err)
```
[the remark this answers is not in the document]
Comment (isseis) [reply test/security/hash_bypass_test.go]: done

File: internal/logging/slack_handler_test.go [hunk 2]
```diff
@@ -351,3 +358,368 @@ func TestNewSlackHandler_URLValidation(t *testing.T) {
...
+		if intervals[i] <= intervals[i-1] {
+			t.Errorf("Expected increasing intervals, but intervals[%d]=%v <= intervals[%d]=%v",
+				i, intervals[i], i-1, intervals[i-1])
+		}
+	}
+
+	// First interval should be at least the base backoff
+	if intervals[0] < backoffBase {
+		t.Errorf("Expected first interval >= %v, got %v", backoffBase, intervals[0])
+	}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/logging/slack_handler_test.go]: done

File: internal/runner/privilege/unix_privilege_test.go [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+	// as it would kill the test process
+
+	// We can't actually call emergencyShutdown as it calls os.Exit
+	// Instead, we verify the function signature is correct
+	assert.NotNil(t, manager)
+
+	// Test that the function exists by checking it's not nil
+	// This is a compile-time check
+	_ = manager.emergencyShutdown
+}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/privilege/unix_privilege_test.go]: done
